SteamTrade/SteamWeb.cs

```diff
@@ -1,336 +1,169 @@
-            byte[] encodedPassword = rsa.Encrypt(bytePassword, false);
-            string encryptedBase64Password = Convert.ToBase64String(encodedPassword);
-
-
-            SteamResult loginJson = null;
-            CookieCollection cookieCollection;
-            string steamGuardText = "";
-            string steamGuardId = "";
-            do
-            {
-                Console.WriteLine("SteamWeb: Logging In...");
-
-                bool captcha = loginJson != null && loginJson.captcha_needed == true;
-                bool steamGuard = loginJson != null && loginJson.emailauth_needed == true;
-
-                string time = Uri.EscapeDataString(rsaJSON.timestamp);
-                string capGID = loginJson == null ? null : Uri.EscapeDataString(loginJson.captcha_gid);
-
-                data = new NameValueCollection();
-                data.Add("password", encryptedBase64Password);
-                data.Add("username", username);
-
-                // Captcha
-                string capText = "";
-                if (captcha)
-                {
-                    Console.WriteLine("SteamWeb: Captcha is needed.");
-                    System.Diagnostics.Process.Start("https://steamcommunity.com/public/captcha.php?gid=" + loginJson.captcha_gid);
-                    Console.WriteLine("SteamWeb: Type the captcha:");
-                    capText = Uri.EscapeDataString(Console.ReadLine());
-                }
-
-                data.Add("captchagid", captcha ? capGID : "");
-                data.Add("captcha_text", captcha ? capText : "");
-                // Captcha end
-
-                // SteamGuard
-                if (steamGuard)
-                {
-                    Console.WriteLine("SteamWeb: SteamGuard is needed.");
-                    Console.WriteLine("SteamWeb: Type the code:");
-                    steamGuardText = Uri.EscapeDataString(Console.ReadLine());
-                    steamGuardId = loginJson.emailsteamid;
-                }
-
-                data.Add("emailauth", steamGuardText);
-                data.Add("emailsteamid", steamGuardId);
-                // SteamGuard end
-
-                data.Add("rsatimestamp", time);
-
-                using(HttpWebResponse webResponse = Request("https://steamcommunity.com/login/dologin/", "POST", data, false))
-                {
-                    using(StreamReader reader = new StreamReader(webResponse.GetResponseStream()))
-                    {
-                        string json = reader.ReadToEnd();
-                        loginJson = JsonConvert.DeserializeObject<SteamResult>(json);
-                        cookieCollection = webResponse.Cookies;
-                    }
-                }
-            } while (loginJson.captcha_needed || loginJson.emailauth_needed);
-
+            Task<WebResponse> responseTask = RequestAsync(url, method, data, ajax, referer);
+            responseTask.Wait();
+            return responseTask.Result as HttpWebResponse;
```

Comment:
`responseTask.Result` already waits for the result, there is no need to call `responseTask.Wait()` as well.